languages/ruby/reference/exercise-concepts/nucleotide-count.md

```diff
@@ -0,0 +1,53 @@
+# Concepts of nucleotide-count
+
+## General
+
+- class level methods: used as the starting point of the exercise. They may be named with `self.[method_name]` notation or with a `self` block. Another way of creating class methods is extending a module.
+- module methods: depending on the approach, they may be used instead of the class methods.
+- factory pattern and object creation: the method `from_dna` may be used to create instances of the main class.
+- instance variables: store the common value/state for the instance.
+- accessors: `att_reader` may be used to access instance variables.
+- access modifiers: `private` may be used to restrict access to the readers.
+- method definition: the `histogram` method needs to be defined.
+- method arguments: the method `from_dna` needs an argument.
+- string: one string is passed as the input.
+- `chars`: may be used to convert the string into an array of characters.
+- strings mutation: `freeze` may be used to prevent string constants from mutating. The directive `frozen_string_literal: true` may also be used for the same purpose.
+- Enumerable module: including `Enumerable` and implementing `each` is one of the possible approaches to this exercise.
+- `Enumerable#count`: one of the possible ways of counting nucleotides.
```

Comment:
I think this is an implementation of a general concept, which is translating a collection (enumerable) to a single value. Not sure how to name this concept.

Reply:
I guess we could group all the Enumerable methods in one concept, something like `Collections operations and transformations` I'm not sure about the naming

Reply:
Me neither. Maybe just do "Collection transformations"?

Reply:
"Iteration" or "enumeration" as a general concept?

Reply:
I think transforming collections is different from iterating them. Or did I read your comment wrong?

Reply:
> I think transforming collections is different from iterating them. Or did I read your comment wrong?

I think that is a fair assessment.  Iterating to transform a collection may be only one way to do so.

Reply:
> (freeze also works on other types, right?).

Freeze can be called without error (on anything?) but doesn't make sense to freeze value objects, such as Numeric, RegExp, Range, Symbol, etc.  That isn't likely to happen here though.